packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx

```diff
@@ -34,15 +34,18 @@ class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps,
 
   requestNextPageAction = (
     { referenceIndex, forceReload },
-    { virtualRows, skip }: Getters,
+    { virtualRows }: Getters,
   ) => {
     const { pageSize, totalRowCount } = this.props;
 
     let newBounds;
     let requestedRange;
     let actualVirtualRows = virtualRows;
     if (forceReload) {
-      newBounds = requestedRange = { start: skip, end: skip + pageSize! * 2 };
+      const loadedInterval = intervalUtil.getRowsInterval(virtualRows);
+      newBounds = requestedRange = getDoublePageInterval(
```

Comment:
loaded interval should be calculated in a single place

Reply:
Done